### PR TITLE
[Woo CYS] Implement `Appearance > Themes` entry point

### PR DESCRIPTION
**Pull request by chihsuan**
Related to #81993 

## Proposed Changes

* Implement Customize Your Store  Appearance > Themes entry point

figma: Mrk6SERPZ4KrFHSjM0a8TK-fi-3388%3A275564

```
Feature: Customize Your Store Intro page when the task is completed

  Scenario: use case 1
    Given user Not yet created a store with AI and the assembler, No changes to any theme (Tsubaki is still the active theme)
    Then it should: 
     1. Display "The Design with AI component" right after the first row.
     2. Remove the “Install theme” button” on the top right.
     3. Align the Upgrade banner with the active color scheme.

  Scenario: use case 2
    Given Not yet created a store with AI and the assembler, Changes to the active theme
    Then it should: 
     1. Display "The Design with AI component" right after the first row.
     2. Display a modal to warn the user we’ll replace content.


  Scenario: use case 2
    Given Already created a store with AI and the assembler, and custom theme is the active theme
    Then it should: 
     1. Show custom theme thumbnail and actions
     2.  Display "The Design with AI component" at the bottom
```
theme-showcase

Screenshot 2023-12-06 at 12 05 43

Screenshot 2023-12-06 at 11 28 37

## Testing Instructions

* Creating a Woo Express Site for Testing CYS pdibGW-2pG-p2
* Change [this](https://github.com/Automattic/wp-calypso/pull/84894/commits/af5d29122f781388fb8e6c2204112e6665c80456#diff-ff4b4ca8173b5696121aac959f8fde70b40dabc0306ef0adfc5f82cd328832faR20) line to return true
* Go to `http://calypso.localhost:3000/themes/<site-url>`
* Confirm 
    1. "The Design with AI component" is displayed right after the first row.
    2. “Install theme” button” on the top right is hidden.
    3.  "Upgrade now" or "Let's get started" Banner should align with the active color scheme.
Screenshot 2023-12-06 at 13 41 56

* Go to `Tools > WCA Test Helper > Features` and enable `customize-store` feature flag.
* Click on `Design with AI` 
* You should be redirected to /wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fdesign-with-ai
* Change your theme
* Click on `Design with AI` 
* A "Start new design" warning model should be displayed at the start.
* Go to WooCommerce > Home
* Completed Customize Your Store task
* Go to `http://calypso.localhost:3000/themes/<site-url>`
* It should show a custom theme thumbnail and actions
* it should display "The Design with AI component" at the bottom
* Please test around 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by dsas — COMMENTED** [review]
When I clicked on "Design with AI" I got redirected to https://deantest82.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fdesign-with-ai which displayed a "Sorry, you are not allowed to access this page." message. 

I then pressed "Finish setup" in the top right corner and I had a list of steps which included "Choose your theme" twice.

Screenshot 2023-12-07 at 14 52 05


I upgraded to the performance plan, and the "Design with AI" button worked, but when I went back to the theme showcase the "Install new theme" button was visible. I'm unsure if this is because I upgraded to performance or because I clicked on the Design with AI button and it did some stuff.

I'm unsure if the number of banners - built by express, design with AI I see on this page is a great experience
Screenshot 2023-12-07 at 15 12 33

It now also says that TT4 is my theme, with an image of the theme which looks nothing like my store. I haven't chosen TT4 and I haven't seen that name before, it feels a bit weird.

If I use 'Design with AI' again, after a modal it reopens the prompt. If I click X to get rid of the AI designer I get sent back to some odd theme gallery: `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fintro` which doesn't look anything like the calypso one I was just in.

File: client/components/theme/index.jsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ export class Theme extends Component {
 			taxonomies: PropTypes.object,
 			update: PropTypes.object,
 			soft_launched: PropTypes.bool,
+			isCustomGeneratedTheme: PropTypes.bool,
```
Comment (dsas) [inline client/components/theme/index.jsx:45]: maybe this prop should be namedspaced harder...we have other kinds of _generated_ themes already, and might have others in the future

File: client/components/theme/index.jsx [hunk 2]
```diff
@@ -143,9 +144,22 @@ export class Theme extends Component {
...
+			this.props;
 		const { description, screenshot } = theme;
 
+		if ( theme.isCustomGeneratedTheme ) {
+			return (
+				<iframe
+					scrolling="no"
+					loading="lazy"
+					title={ translate( 'Custom Theme Preview' ) }
+					className="theme__site-preview"
```
Comment (dsas) [inline client/components/theme/index.jsx:159]: the classname here is perhaps also a bit of a grab, should it have `cys` or `woo` in it?

File: client/components/themes-list/index.jsx [hunk 3]
```diff
@@ -95,6 +119,60 @@ export const ThemesList = ( { tabFilter, ...props } ) => {
...
+		}
+
+		const theme = themes[ activeThemeIndex ];
+
+		return [
+			...themes.slice( 0, activeThemeIndex ),
+			{
+				...theme,
+				isCustomGeneratedTheme: true,
+				name: translate( 'My custom theme' ),
```
Comment (dsas) [inline client/components/themes-list/index.jsx:164]: Where is this displayed? I'm not seeing it 🤔

File: client/components/woo-design-with-ai-warning-modals/index.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+		<Modal
+			className={ classname }
+			title={ __( 'Are you sure you want to start over?', 'woocommerce' ) }
+			onRequestClose={ () => setOpenModal( false ) }
+			shouldCloseOnClickOutside={ false }
+		>
+			<p>
+				{ createInterpolateElement(
+					__(
+						"You'll be asked to provide your business info again, and will lose your existing AI design. If you want to customize your existing design, you can do so via the <EditorLink>Editor</EditorLink>.",
```
Comment (dsas) [inline client/components/woo-design-with-ai-warning-modals/index.tsx:77]: it feels a bit bad that we lose text the user has painstakingly entered - perhaps they just want to tweak it a bit.

**Comment by chihsuan** [comment]
Hey @dsas Thank you so much for reviewing my PR and great feedback!  🙂 

> When I clicked on "Design with AI" I got redirected to https://deantest82.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fdesign-with-ai which displayed a "Sorry, you are not allowed to access this page." message.
> 
> I then pressed "Finish setup" in the top right corner and I had a list of steps which included "Choose your theme" twice.
> 
> Screenshot 2023-12-07 at 14 52 05

Ah, the feature flag needs to be turned on before testing to see the customize your store pages. 😓  I've updated the instructions!

> I upgraded to the performance plan, and the "Design with AI" button worked, but when I went back to the theme showcase the "Install new theme" button was visible. I'm unsure if this is because I upgraded to performance or because I clicked on the Design with AI button and it did some stuff.

The "Install new theme" button should be hidden when the theme is the default WooExpress theme (Tsubaki) but not TT4, so this is expected.

> I'm unsure if the number of banners - built by express, design with AI I see on this page is a great experience Screenshot 2023-12-07 at 15 12 33

👍 

> It now also says that TT4 is my theme, with an image of the theme which looks nothing like my store. I haven't chosen TT4 and I haven't seen that name before, it feels a bit weird.

The theme would be switched to TT4 when the AI flow is complete. 

We intend to replace TT4 with a custom theme card when AI generates it in this PR. It's weird that you still see the TT4.

However, we noticed that it took awhile (~1-2 mins) for the wpcom API to update the response. I suspected you saw the TT4 because of that. 🤔 

> If I use 'Design with AI' again, after a modal it reopens the prompt. If I click X to get rid of the AI designer I get sent back to some odd theme gallery: `/wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store%2Fintro` which doesn't look anything like the calypso one I was just in.

Good point! I'll forward this feedback to our designer.

File: client/components/themes-list/index.jsx (hunk 3 above)
Comment (chihsuan) [reply client/components/themes-list/index.jsx:164]: It's displayed when the theme is generated with CYS AI flow, and custom theme is the active theme.

Untitled

File: client/components/woo-design-with-ai-warning-modals/index.tsx (hunk 4 above)
Comment (chihsuan) [reply client/components/woo-design-with-ai-warning-modals/index.tsx:77]: Yep, this is exactly why we show the warn modal. 😅

> If you want to customize your existing design, you can do so via the <EditorLink>Editor</EditorLink>

They can go to site editor to tweak their theme.

File: client/components/theme/index.jsx (hunk 2 above)
Comment (chihsuan) [reply client/components/theme/index.jsx:159]: This is not very woo-specific. I think we can set `isCustomGeneratedTheme` prop to true for any custom theme and use this to display a live preview as a theme image. What do you think? 🙂

Comment (dsas) [reply client/components/theme/index.jsx:159]: That's true but the GET vars to the iframe src are CYS specific I think....maybe that isn't something to worry about right now

**Comment by dsas** [comment]
> However, we noticed that it took awhile (~1-2 mins) for the wpcom API to update the response. I suspected you saw the TT4 because of that. 

I'm still seeing TT4 now.. It's listed as the active theme in both wp-admin/themes.php as well as in calypso. If I look at the front of the site, it's definitely a custom TT4 template and possibly with a styling problem..

**Comment by chihsuan** [comment]
> I'm still seeing TT4 now.. It's listed as the active theme in both wp-admin/themes.php as well as in calypso. If I look at the front of the site, it's definitely a custom TT4 template and possibly with a styling problem..

Hey @dsas I've taken a look at your site. I added `test_add_woocommerce_options_to_jetpack_sync` filter (D131047-code)  via the `Code Snippets` plugin and it now works! 🙌 


Screenshot 2023-12-08 at 17 19 42

**Comment by dsas** [comment]
Doh! Thanks for fixing that, it's quite the complicated set of test instructions!

**Review by rjchow — APPROVED** [review]
tests well, thanks!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10647467

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @chihsuan for including a screenshot in the description! This is really helpful for our translators.

File: client/my-sites/themes/theme-options.js [hunk 5]
```diff
@@ -328,6 +328,31 @@ function getAllThemeOptions( { translate, isFSEActive } ) {
...
+			label: translate( 'Quick editing in the Assembler' ),
+			extendedLabel: translate( 'Quick editing in the Assembler' ),
+			getUrl: () => {
+				return `${ siteAdminUrl }admin.php?page=wc-admin&path=%2Fcustomize-store%2Fassembler-hub&customizing=true`;
+			},
+		},
+		customize: {
+			...options.customize,
+			label: translate( 'Advance customization in the Editor' ),
+			extendedLabel: translate( 'Advance customization in the Editor' ),
```
Comment (emilyaudela) [inline client/my-sites/themes/theme-options.js:344]: 👋 Should this be "Advanced" (as an adjective) instead of "Advance" (verb)?

Comment (chihsuan) [reply client/my-sites/themes/theme-options.js:344]: Thanks for flagging this! Confirming with the team. 🙏

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10647467).

**Pull request merged**